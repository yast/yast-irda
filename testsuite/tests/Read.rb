# encoding: utf-8

#  Read.ycp
#  Test of Irda::Read function
#  Author: Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class ReadClient < Client
    def main
      # testedfiles: Irda.ycp Service.ycp
      Yast.import "Testsuite"
      Yast.import "Irda"

      @READ = {
        "init"   => { "scripts" => { "exists" => true } },
        # no sysconfig file
        "target" => { "stat" => {} }
      }

      Testsuite.Dump("==== reading... ============================")

      Testsuite.Test(lambda { Irda.Read }, [@READ, {}, {}], 0)

      Testsuite.Dump("============================================")

      @READ = {
        "init"      => { "scripts" => { "exists" => true } },
        "target"    => { "stat" => { "isreg" => true } },
        "sysconfig" => {
          "irda" => { "IRDA_PORT" => "/dev/ttyS1", "IRDA_MAX_BAUD_RATE" => nil }
        }
      }

      Testsuite.Test(lambda { Irda.Read }, [@READ, {}, {}], 0)

      nil
    end
  end
end

Yast::ReadClient.new.main
