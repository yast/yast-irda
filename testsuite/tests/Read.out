Dump	==== reading... ============================
Return	true
Dump	============================================
Read	.sysconfig.irda.IRDA_PORT "/dev/ttyS1"
Read	.sysconfig.irda.IRDA_MAX_BAUD_RATE nil
Execute	.target.bash "/bin/systemctl status irda.service" $["TERM":"raw"] 0
Return	true
